directories/src/redux/epics/treasure-hunt-epics.js

```diff
@@ -0,0 +1,241 @@
+    });
+};
+
+const requestBrokerEpic = (action$, store) => {
+  return action$.ofType(nodeActions.NODE_REQUEST_BROKER_NODES).mergeMap(() => {
+    const { brokerNodes } = store.getState().node;
+    const currentList = brokerNodes.map(bn => bn.address);
+    return Observable.fromPromise(
+      brokerNode.requestBrokerNodeAddressPoW(currentList)
+    )
+      .mergeMap(({ data }) => {
+        const { id: txid, pow: { message, address, branchTx, trunkTx } } = data;
+
+        // TODO: change this
+        const value = 0;
+        const tag = "EDMUNDANDREBELWUZHERE";
+        const seed =
+          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+
+        return Observable.fromPromise(
+          iota.prepareTransfers({ address, message, value, tag, seed })
+        ).map(trytes => {
+          return { txid, trytes, branchTx, trunkTx };
+        });
+      })
+      .mergeMap(({ txid, trytes, branchTx, trunkTx }) =>
+        Observable.fromPromise(
+          iota.attachToTangleCurl({
+            branchTx,
+            trunkTx,
+            mwm: 14,
+            trytes
+          })
+        ).map(trytesArray => {
+          return { txid, trytesArray };
+        })
+      )
+      .mergeMap(({ txid, trytesArray }) =>
+        Observable.fromPromise(
+          brokerNode.completeBrokerNodeAddressPoW(txid, trytesArray[0])
+        ).mergeMap(({ data }) => {
+          const { purchase: address } = data;
+          return [
+            nodeActions.addBrokerNode({ address }),
+            nodeActions.determineBrokerNodeOrGenesisHash()
+          ];
+        })
+      )
+      .catch(error => {
+        console.log("BROKER NODE ADDRESS FETCH ERROR", error);
+        return Observable.empty();
+      });
+  });
+};
+
+const requestGenesisHashEpic = (action$, store) => {
+  return action$
+    .ofType(nodeActions.NODE_REQUEST_GENESIS_HASHES)
+    .mergeMap(() => {
+      const { newGenesisHashes } = store.getState().node;
+      const currentList = newGenesisHashes.map(gh => gh.genesisHash);
+      return Observable.fromPromise(
+        brokerNode.requestGenesisHashPoW(currentList)
+      )
+        .mergeMap(({ data }) => {
+          const {
+            id: txid,
+            pow: { message, address, branchTx, trunkTx }
+          } = data;
+
+          // TODO: change this
+          const value = 0;
+          const tag = "EDMUNDANDREBELWUZHERE";
+          const seed =
+            "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+
+          return Observable.fromPromise(
+            iota.prepareTransfers({ address, message, value, tag, seed })
+          ).map(trytes => {
+            return { txid, trytes, branchTx, trunkTx };
+          });
+        })
+        .mergeMap(({ txid, trytes, branchTx, trunkTx }) =>
+          Observable.fromPromise(
+            iota.attachToTangleCurl({
+              branchTx,
+              trunkTx,
+              mwm: 14,
```

Comment:
Don't worry about it for this PR, but sometime this week we'll want to change all these mwm: 14 to mwm: 9, once we have all the test brokers on the private tangle.